ws/filetransfer/model.go

```diff
@@ -40,38 +54,33 @@ type GetFile struct {
 	Path string `msgpack:"path,omitempty" json:"path,omitempty"`
 }
 
-// Put file marks the flow of MessageTypeFileChunk messages to be started to the remote end
-type PutFile struct {
-	// The file path to the file we are requesting
-	Path string `msgpack:"path,omitempty" json:"path,omitempty"`
-}
-
 // Stat file requests the file stat structure from the remote end
 type StatFile struct {
 	// The file path to the file we are requesting
 	Path string `msgpack:"path,omitempty" json:"path,omitempty"`
 }
 
-// Stat file response is the reply to the StatFile from the remote end
-type StatFileResponse struct {
+// FileInfo is the object returned from a StatFile request and is also used
+// for "put_file" requests for specifying the target file.
+type FileInfo struct {
 	// The file path to the file we are sending status for
 	Path string `msgpack:"path,omitempty" json:"path,omitempty"`
 	// The file size
 	Size int64 `msgpack:"size,omitempty" json:"size,omitempty"`
 	// The file type
 	Type string `msgpack:"type,omitempty" json:"type,omitempty"`
 	// The file owner
-	Owner string `msgpack:"owner,omitempty" json:"owner,omitempty"`
+	UID uint32 `msgpack:"owner,omitempty" json:"owner,omitempty"`
 	// The file group
-	Group string `msgpack:"group,omitempty" json:"group,omitempty"`
+	GID uint32 `msgpack:"group,omitempty" json:"group,omitempty"`
```

Comment:
nitpick: UserID and GroupID would probably fit the naming better.

Reply:
I thought that string username and string group would be more portable, if the UI wants to show some info it would need another call to resolve the user and group -- the ids will not tell much to the UI user.

Reply:
The only problem is that the filesystem is not aware of named users/groups, and I would assume that filetransfer would preserve the inode UID/GID by default (as `scp` does).